src/db/mysql/models.rs

```diff
@@ -343,7 +345,9 @@ impl MysqlDb {
         .optional()?
         .ok_or(DbErrorKind::CollectionNotFound)?
         .id;
-        self.coll_cache.put(id, name.to_owned())?;
+        if !self.session.borrow().in_transaction {
```

Comment:
Instead of guarding this with `in_transaction`, we need the equivalent of the Spanner `models.rs` where we denote whether or not we're in a write transaction.

We'll know we're writing if a `lock_for_write_sync` was called, so it should toggle a new `in_write_transaction` flag that we can check instead.